node/pkg/driver/node_utils.go

```diff
@@ -425,7 +425,8 @@ func (n NodeUtils) GenerateNodeID(hostName string, fcWWNs []string, iscsiIQN str
 func (n NodeUtils) GetTopologyLabels(ctx context.Context, nodeName string) (map[string]string, error) {
 	kubeConfig, err := rest.InClusterConfig()
 	if err != nil {
-		return nil, err
+		logger.Infof("unable to load in-cluster configuration: %v", err)
+		return nil, nil
```

Comment:
why is that OK? (no err returned)

Reply:
for e.g. community tests

Reply:
What's special with the community tests in this case? They don't need to get the error?

Reply:
they fail if we error, since they check our CSI spec compliance, not our k8s compatibility.
even if we currently officially support only k8s-based orchestrators, our code should still be spec compliant, i.e. not fail just because it's outside k8s

Reply:
I see.
The code doesn't explain this situation very well. I'm not a big fan of creating a variable called `csi_spec_expects_no error_in_this_case = nil;` and use it where needed, but I don't like the current code. WDYT?
Also, should the log severity be higher? Elaborate on the relevant errors, please.

Reply:
I think the log severity tells that this external error is not considered an error on our side, which saves the need for the suggested variable

Reply:
@ArbelNathan, WDYT?

Reply:
I can add another log message `we are probably not in k8s, skipping topology retrieval`

Reply:
Assuming things in logs can lead to questions when it happens in prod. env.
May I suggest adding `skipping topology retrieval. Are we in K8s?`?

Reply:
I don't see a point for logs to ask rhetorical questions, could we settle for an additional `skipping topology retrieval` message?

Reply:
sure

Reply:
done

Reply:
@ArbelNathan could this conversation be resolved?